pdr_backend/lake/csv_data_store.py

```diff
@@ -0,0 +1,417 @@
+import os
+from typing import List, Optional, Union
+import polars as pl
+from polars.type_aliases import SchemaDict
+
+from enforce_typing import enforce_types
+from pdr_backend.lake.base_data_store import BaseDataStore
+
+
+class CSVDataStore(BaseDataStore):
+
+    @enforce_types
+    def _get_folder_path(self, dataset_identifier: str) -> str:
+        """
+        Returns the folder path for the given dataset_identifier.
+        If the folder does not exist, it will be created.
+        @args:
+            dataset_identifier: str - identifier of the dataset
+        """
+        folder_path = os.path.join(self.base_path, dataset_identifier)
+        os.makedirs(folder_path, exist_ok=True)
+        return folder_path
+
+    @enforce_types
+    def _fill_with_zero(self, number: int, length: int = 10) -> str:
```

Comment:
this is not **filling**, it is **padding**. Please adjust the name of this function.
You can also use something MUCH more simple, e.g. zfill, rjust or number formatting with f:

https://stackoverflow.com/questions/339007/how-do-i-pad-a-string-with-zeros